homeassistant/components/ecoforest/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""The Ecoforest integration."""
+from __future__ import annotations
+
+import logging
+
+import async_timeout
+import httpx
+from pyecoforest.api import EcoforestApi
+from pyecoforest.exceptions import EcoforestAuthenticationRequired
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import EcoforestCoordinator
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Ecoforest from a config entry."""
+
+    host = entry.data[CONF_HOST]
+    auth = httpx.BasicAuth(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
+    api = EcoforestApi(host, auth)
+
+    try:
+        async with async_timeout.timeout(10):
+            device = await api.get()
+            _LOGGER.debug("Ecoforest: %s", device)
+    except EcoforestAuthenticationRequired:
+        _LOGGER.error("Authentication on device (%s)  failed", host)
+        return False
+    except Exception:  # pylint: disable=broad-except
+        _LOGGER.error("Error communicating with device %s", host)
+        return False
```

Comment:
You can raise ConfigEntryNotReady here.

Reply:
I have used false, should I raise the exception both catches or just in the broad?

Reply:
Just in the broad one, since when someone's credentials are expired we should stop trying, when it reached the broad we can continue trying later on